proc/threads.c

```diff
@@ -1275,88 +1353,201 @@ int threads_sigpost(process_t *process, thread_t *thread, int sig)
  */
 
 
-int _proc_lockSet(lock_t *lock, int interruptible, spinlock_ctx_t *sc)
+static unsigned int _proc_lockGetPriority(lock_t *lock)
 {
-	while (lock->v == 0) {
-		if (proc_threadWaitEx(&lock->queue, &lock->spinlock, 0, interruptible, sc) == -EINTR)
+	unsigned int ret = -1;
+	thread_t *t;
+
+	if ((t = lock->queue) != NULL) {
```

Comment:
MISRA

Reply:
and other instances of if ((var = func()) != const)

Reply:
Done.